### DISCUSSION TITLE
DICOM tag validation

### DISCUSSION THREAD

**Discussion by janossch** [dcm4che/dcm4che#1142] [Q&A]
Dear dcm4che Community,

I'm wondering if are there any built-in possibility to validate the incoming data (actually each or given tags) based on its [value representation](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html#table_6.2-1). I know data elements within an Attributes (a single dataset) can be [validated](https://github.com/dcm4che/dcm4che/blob/d60ddcf636202e396aa55f41b1df2c6cf3186c4c/dcm4che-core/src/main/java/org/dcm4che3/data/Attributes.java#L3355) against predefined IODs. But AFAIK it "just" checks [data element types](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_7.4.html).

e.g.:
let's suppose we extended [BasicCStoreSCP](https://github.com/dcm4che/dcm4che/blob/d60ddcf636202e396aa55f41b1df2c6cf3186c4c/dcm4che-net/src/main/java/org/dcm4che3/net/service/BasicCStoreSCP.java#L76) and would like to further process the _AffectedSOPInstanceUID_ from the request, but we would like to make sure it conforms to **VR.UI**. The _AffectedSOPInstanceUID_ is a **VR.UI** (_String_ in Java) and it must contain only "0"-"9", "." of Default Character Repertoire. 

Are there any possiblitiy in the dcm4che library to do such validation on given tags or even in the whole _Attributes_ instance?
In other words can dcm4che validate tags based on their VR?

Thank you in advance!
Best Regards
